src/plots/linePlot.ts

```diff
@@ -349,10 +373,59 @@ export module Plots {
 
     protected _getDataToDraw() {
       let dataToDraw = new Utils.Map<Dataset, any[]> ();
-      this.datasets().forEach((dataset) => dataToDraw.set(dataset, [dataset.data()]));
+
+      this.datasets().forEach((dataset) => {
+        let data = dataset.data();
+
+        if (!this._croppedRendering) {
+          dataToDraw.set(dataset, [data]);
+          return;
+        }
+
+        let filteredDataIndices = data.map((d, i) => i);
```

Comment:
I am not sure, but you didn't use `indices` below to do any complex stuff
isn't `indices[i]` just `i`, `indices[i - 1]`  just `i - 1`, etc..? 
and `indices.length` just `data.length` ?